packages/test/test-service-load/src/loadTestDataStore.ts

```diff
@@ -473,31 +484,56 @@ class LoadTestDataStore extends DataObject implements ILoadTest {
+        const opSizeinBytes = (typeof config.testConfig.opSizeinBytes === 'undefined') ? 
+        0 : config.testConfig.opSizeinBytes;
+        assert(opSizeinBytes >= 0, "opSizeinBytes must be greater than or equal to zero.");
+        const generateStringOfSize = (sizeInBytes: number): string => new Array(sizeInBytes + 1).join("0");
+        let opsSent = 0; 
         try {
-            while (dataModel.counter.value < clientSendCount && !this.disposed) {
-                // this enables a quick ramp down. due to restart, some clients can lag
-                // leading to a slow ramp down. so if there are less than half the clients
-                // and it's partner is done, return true to complete the runner.
-                if (this.runtime.getAudience().getMembers().size < config.testConfig.numClients / 2
-                    && ((await dataModel.getPartnerCounter())?.value ?? 0) >= clientSendCount) {
-                    return true;
+            if (opSizeinBytes === 0) {
+                while (dataModel.counter.value < clientSendCount && !this.disposed) {
+                    // this enables a quick ramp down. due to restart, some clients can lag
+                    // leading to a slow ramp down. so if there are less than half the clients
+                    // and it's partner is done, return true to complete the runner.
+                    if (this.runtime.getAudience().getMembers().size < config.testConfig.numClients / 2
+                        && ((await dataModel.getPartnerCounter())?.value ?? 0) >= clientSendCount) {
+                        return true;
+                    }
+                    if (dataModel.assigned()) {
+                        dataModel.counter.increment(1);
+                        opsSent = dataModel.counter.value;
+                        if (opsSent % opsPerCycle === 0) {
+                            await dataModel.blobFinish();
+                            dataModel.abandonTask();
+                            // give our partner a half cycle to get the task
+                            await delay(cycleMs / 2);
+                        } else {
+                            // Random jitter of +- 50% of opWaitMs
+                            await delay(opsGapMs + opsGapMs * random.real(0, .5, true)(config.randEng));
+                        }
+                    } else {
+                        await dataModel.volunteerForTask();
+                    }
                 }
-
-                if (dataModel.assigned()) {
-                    dataModel.counter.increment(1);
-                    if (dataModel.counter.value % opsPerCycle === 0) {
-                        await dataModel.blobFinish();
-                        dataModel.abandonTask();
-                        // give our partner a half cycle to get the task
-                        await delay(cycleMs / 2);
+            } else {
+                while (opsSent < clientSendCount && !this.disposed) {
```

Comment:
seems like this variant will take twice as long to run, since condition in the other loop is for the counter state (which is shared between this client and its partner client).

Reply:
as long as you all are aware of this I don't have an issue with it. Just wanted to point it out so there isn't a question of "why is it taking much longer" after this gets merged.